infra/main.json

```diff
@@ -4,8 +4,8 @@
   "metadata": {
     "_generator": {
       "name": "bicep",
-      "version": "0.26.170.59819",
-      "templateHash": "12294581750482242115"
+      "version": "0.23.1.45101",
```

Comment:
nit: you might want to update your AZ CLI, that might also help fixing the various warnings thrown when compiling to JSON

Reply:
I was building with `bicep build main.bicep`
you are using `az bicep build --file main.bicep`

I removed all warnings except one.  It was not related to compiler but how I had bicep written.

Reply:
yeah I didn't want to install yet another tool, it seems counterintuitive considering how much effort has been put into `az`